### ISSUE TITLE
question about training dataset. 

### ISSUE THREAD

**Issue by cactusgame** [ZZZHANG-jx/GCDRNet#3]
Hi, thanks for your great paper, I have a question when I re-produce your work in the paper. 

As mentioned in the paper that 

> Owing to the lightweight structure of our backbone, DR-Net takes the original source image as input without down-sampling or cropping, making our enhanced results free from splicing traces. For Is with resolutions below 512, we resized the short side to 512 while maintaining the aspect ratio

Do you mean that the pre-trained `DRNet` is trained only by the `RealDAE` dataset? You mentioned the batch size is 16 during the training, without the data cropping, so there are only 600/16 = 38 iterations an epoch to train the `DRNet`? 

Could you tell me how many epochs you need to train the pre-trained DRNet? Currently, I find that it's hard to reproduce your result only with the RealDAE dataset. Or am I misunderstanding something?

**Comment by ZZZHANG-jx** [comment]
> Hi, thanks for your great paper, I have a question when I re-produce your work in the paper.
> 
> As mentioned in the paper that
> 
> > Owing to the lightweight structure of our backbone, DR-Net takes the original source image as input without down-sampling or cropping, making our enhanced results free from splicing traces. For Is with resolutions below 512, we resized the short side to 512 while maintaining the aspect ratio
> 
> Do you mean that the pre-trained `DRNet` is trained only by the `RealDAE` dataset? You mentioned the batch size is 16 during the training, without the data cropping, so there are only 600/16 = 38 iterations an epoch to train the `DRNet`?
> 
> Could you tell me how many epochs you need to train the pre-trained DRNet? Currently, I find that it's hard to reproduce your result only with the RealDAE dataset. Or am I misunderstanding something?

Thank you for your interest in our work!

Training directly with real images can indeed be challenging in terms of stability, as manually labeled data often carries a degree of subjectivity, leading to inconsistent annotations. We encountered similar issues in our research, which led us to adopt a progressive learning approach, initially focusing on data with more consistent annotations (such as those without colored figures).

To further enhance training stability, you might consider using synthetic data for pre-training. In our latest work, we found that training solely with synthetic data not only converges more quickly but also yields quite effective results. You can refer to the following link to construct training data based on Doc3DShade and train DRNet and GCNet: https://github.com/ZZZHANG-jx/DocRes/tree/master/data#appearance-enhancement

**Comment by cactusgame** [comment]
Thanks for your suggestion, I'll have a try, thanks!

> To further enhance training stability, you might consider using synthetic data for pre-training. In our latest work, we found that training solely with synthetic data not only converges more quickly but also yields quite effective results. You can refer to the following link to construct training data based on Doc3DShade and train DRNet and GCNet: https://github.com/ZZZHANG-jx/DocRes/tree/master/data#appearance-enhancement

**Comment by cactusgame** [comment]
Hi, following the document mentioned, I synthesized 10,000 pairs of training images using Doc3dShade, and trained both gcnet and drnet for 50 epochs each.
As you said, the training converges very quickly, but the evaluation metrics are unsatisfactory. Could you provide suggestions on achieving better training results for GCDRNet?

Additionally, I understand that DocRes performs better, but I am looking for a lighter model. If you have any better recommendations regarding lightweight models, I would be very grateful to hear them. Thank you very much.

The picture below is the validation metrics for DRNet.
![image](https://github.com/user-attachments/assets/adcd0930-b8c3-4d35-9cc3-29545e7ad015)

**Comment by cactusgame** [comment]
list some validation results here.  
- The 1st column is the input degraded image
- The 2nd column is the ground truth image
- The 3rd column is the prediction by GCDRNet

![image](https://github.com/user-attachments/assets/b9091da3-cf46-4566-bac6-035547c00576)

**Comment by cactusgame** [comment]
Another issue is that some images exhibit a "dirty" effect at the edges, similar to the one shown below. Do you happen to know how to address this kind of issue?

![image](https://github.com/user-attachments/assets/4529f17f-877f-4822-a264-ca592d0a69aa)

**Comment by ZZZHANG-jx** [comment]
> Hi, following the document mentioned, I synthesized 10,000 pairs of training images using Doc3dShade, and trained both gcnet and drnet for 50 epochs each. As you said, the training converges very quickly, but the evaluation metrics are unsatisfactory. Could you provide suggestions on achieving better training results for GCDRNet?
> 
> Additionally, I understand that DocRes performs better, but I am looking for a lighter model. If you have any better recommendations regarding lightweight models, I would be very grateful to hear them. Thank you very much.
> 
> The picture below is the validation metrics for DRNet. ![image](https://private-user-images.githubusercontent.com/5027167/363455484-adcd0930-b8c3-4d35-9cc3-29545e7ad015.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUyODM1ODAsIm5iZiI6MTcyNTI4MzI4MCwicGF0aCI6Ii81MDI3MTY3LzM2MzQ1NTQ4NC1hZGNkMDkzMC1iOGMzLTRkMzUtOWNjMy0yOTU0NWU3YWQwMTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDJUMTMyMTIwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWE4MzY1ZmExZTJjMzc4NmEyMjBmNjkwM2IwOTA4OTQ4NzY0ZmI1MGMzYjMzOTQ4ZjI1Nzk5NmRhNDY2NmIxMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.OoIAjBH12dmxoxD7auoXEOVyf1tQ7aPH_rEZ-g-elts)

From your results, it seems that there is an issue with poor color information recovery, likely due to a lack of similar data. I suggest adding some PDF data similar to magazines that contain more figures to help the model learn color recovery better. Other suggestions include: first, train GCNet, then fix GCNet and train DRNet (consider adding real data at this stage), and finally, consider joint training of GCNet and DRNet with a reduced learning rate. You can start with the simplest L1 loss and single scale.

**Comment by cactusgame** [comment]
Thank you for your suggestions; they are accommodating!

I have one last question. You mentioned that "You can start with the simplest L1 loss and single scale" Is it okay to only introduce multi-scale training in the final stage (joint training of GCNet and DRNet)? Do you mean that the excessive shadows appearing at the edges, as shown in the image above, are likely due to a lack of multi-scale training?

> From your results, it seems that there is an issue with poor color information recovery, likely due to a lack of similar data. I suggest adding some PDF data similar to magazines that contain more figures to help the model learn color recovery better. Other suggestions include: first, train GCNet, then fix GCNet and train DRNet (consider adding real data at this stage), and finally, consider joint training of GCNet and DRNet with a reduced learning rate. You can start with the simplest L1 loss and single scale.

**Comment by ZZZHANG-jx** [comment]
> Thank you for your suggestions; they are accommodating!
> 
> I have one last question. You mentioned that "You can start with the simplest L1 loss and single scale" Is it okay to only introduce multi-scale training in the final stage (joint training of GCNet and DRNet)? Do you mean that the excessive shadows appearing at the edges, as shown in the image above, are likely due to a lack of multi-scale training?
> 
> > From your results, it seems that there is an issue with poor color information recovery, likely due to a lack of similar data. I suggest adding some PDF data similar to magazines that contain more figures to help the model learn color recovery better. Other suggestions include: first, train GCNet, then fix GCNet and train DRNet (consider adding real data at this stage), and finally, consider joint training of GCNet and DRNet with a reduced learning rate. You can start with the simplest L1 loss and single scale.

It is okay to introduce multi-scale training only in the final stage. Even with multi-scale training, the issue you mentioned could still occur, though multi-scale training might help mitigate the problem.

**Comment by NakupendaAI** [comment]
> 请在此列出一些验证结果。
> 
> * 第一列是输入的退化图像。
> * 第二列是真实图像。
> * 第三列是GCDRNet的预测结果。
> 
> ![image](https://github.com/user-attachments/assets/b9091da3-cf46-4566-bac6-035547c00576)

在中国，很多大学里的项目都是为了完成毕业论文而已，你还真相信这些模型可以应用到真实场景么
